lntest/itest/lnd_onchain_test.go

```diff
@@ -379,3 +382,237 @@ func testAnchorReservedValue(net *lntest.NetworkHarness, t *harnessTest) {
+	// commitment transaction and anchor sweep in the mempool.
+	const expectedTxns = 2
+	sweepTxns, err := getNTxsFromMempool(
+		net.Miner.Client, expectedTxns, minerMempoolTimeout,
+	)
+	require.NoError(t.t, err, "no sweep txns in miner mempool")
+	aliceCloseTx := waitingClose.Commitments.LocalTxid
+	_, aliceAnchor := findCommitAndAnchor(t, net, sweepTxns, aliceCloseTx)
+
+	// We'll now mine _only_ the commitment force close transaction, as we
+	// want the anchor sweep to stay unconfirmed.
+	var emptyTime time.Time
+	forceCloseTxID, _ := chainhash.NewHashFromStr(aliceCloseTx)
+	commitTxn, err := net.Miner.Client.GetRawTransaction(
+		forceCloseTxID,
+	)
+	if err != nil {
+		t.Fatalf("unable to get transaction: %v", err)
+	}
+	_, err = net.Miner.GenerateAndSubmitBlock(
+		[]*btcutil.Tx{commitTxn}, -1, emptyTime,
+	)
+	if err != nil {
+		t.Fatalf("unable to generate block: %v", err)
+	}
+
+	// With the anchor output located, and the main commitment mined we'll
+	// instruct the wallet to send all coins in the wallet to a new address
+	// (to the miner), including unconfirmed change.
+	minerAddr, err := net.Miner.NewAddress()
+	if err != nil {
+		t.Fatalf("unable to create new miner addr: %v", err)
+	}
+	sweepReq := &lnrpc.SendCoinsRequest{
+		Addr:             minerAddr.String(),
+		SendAll:          true,
+		MinConfs:         0,
+		SpendUnconfirmed: true,
+	}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	sweepAllResp, err := alice.SendCoins(ctxt, sweepReq)
+	if err != nil {
+		t.Fatalf("unable to sweep coins: %v", err)
+	}
+
+	// Next, we'll shutdown Alice, and allow 16 blocks to pass so that the
+	// anchor output can be swept by anyone. Rather than use the normal API
+	// call, we'll generate a series of _empty_ blocks here.
+	aliceRestart, err := net.SuspendNode(alice)
+	if err != nil {
+		t.Fatalf("unable to shutdown alice: %v", err)
+	}
+	const anchorCsv = 16
+	for i := 0; i < anchorCsv; i++ {
+		_, err := net.Miner.GenerateAndSubmitBlock(nil, -1, emptyTime)
+		if err != nil {
+			t.Fatalf("unable to generate block: %v", err)
+		}
+	}
+
+	// Before we sweep the anchor, we'll restart Alice.
+	//
+	// TODO(roasbeef): should do this after we sweep?
+	if err := aliceRestart(); err != nil {
+		t.Fatalf("unable to restart alice: %v", err)
+	}
+
+	// Now that the channel has been closed, and Alice has an unconfirmed
+	// transaction spending the output produced by her anchor sweep, we'll
+	// mine a transaction that double spends the output.
+	thirdPartyAnchorSweep := genAnchorSweep(t, net, aliceAnchor, anchorCsv)
+	_, err = net.Miner.GenerateAndSubmitBlock(
+		[]*btcutil.Tx{thirdPartyAnchorSweep}, -1, emptyTime,
+	)
+	if err != nil {
+		t.Fatalf("unable to generate block: %v", err)
+	}
+
+	// At this point, we should no longer find Alice's transaction that
+	// tried to sweep the anchor in her wallet.
+	assertTransactionNotInWallet(t.t, alice, aliceAnchor.SweepTx.TxHash())
```

Comment:
Should we assert that the sweep tx is in our wallet + spending the anchor before we restart?

Reply:
Def, added!